src/main/java/knightminer/inspirations/utility/block/BlockCarpetedPressurePlate.java

```diff
@@ -36,8 +41,19 @@ public VoxelShape getShape(@Nonnull BlockState state, @Nonnull IBlockReader worl
 		return flag ? PRESSED_AABB : UNPRESSED_AABB;
 	}
 
+	// Since fitted carpets may replace the original carpet, we need to lookup the item to make sure we get the right
+	// object. So defer until the first time we actually are pick-blocked.
+	private Item pickItem = Items.AIR;
+
 	@Override
 	public ItemStack getPickBlock(BlockState state, RayTraceResult target, IBlockReader world, BlockPos pos, PlayerEntity player) {
-		return new ItemStack(carpet);
+		if (pickItem == Items.AIR) {
+			pickItem = ForgeRegistries.ITEMS.getValue(new ResourceLocation(color.getTranslationKey() + "_carpets"));
+		}
+		if(pickItem == Items.AIR) {
```

Comment:
Nest this in the previous `if`, so we don't have to run the check twice when we already have the carpet